packages/mui-base/src/useMenu/useMenu.ts

```diff
@@ -52,7 +52,10 @@ export default function useMenu(parameters: UseMenuParameters = {}): UseMenuRetu
   const stateChangeHandler: StateChangeCallback<MenuInternalState> = React.useCallback(
     (event, field, fieldValue, reason, state) => {
       if (field === 'open') {
-        onOpenChange?.(fieldValue as boolean);
+        onOpenChange?.(
+          event as Parameters<NonNullable<UseMenuParameters['onOpenChange']>>[0],
```

Comment:
since `event` type is quite long (React.MouseEvent<Element, Event> | React.KeyboardEvent<Event> | React.FocusEvent | null) . I'm reuisng event type from (https://github.com/mui/material-ui/pull/36917/files#diff-e0fe27842c8f1640f1be871ebde00a7da6655afe3da99d11a6b139d3914ca3ed) where ever possible